src/utils.ts

```diff
@@ -91,13 +106,45 @@ export function intersectMany<T>(arrays: T[][]): T[] {
     }
   }
 
-  return arrays[0].filter(elem => {
-    const count = set.get(elem);
-    if (count !== undefined) {
-      set.set(elem, 0);
+  const result: TokenScore[] = [];
+
+  for (const [token, [count, score]] of set) {
+    if (count === arrLength) {
+      result.push([token, score]);
     }
-    return count === arrays.length;
-  });
+  }
+
+  return result;
+}
+
+export function insertSortedValue<T = unknown>(arr: T[], el: T, compareFn?: (a: T, b: T) => number): T[] {
+  if (!compareFn) {
+    compareFn = (a, b) => {
```

Comment:
To allow inlining, extract this function somewhere and then reference it, rather than creating it every time.

Reply:
Moreover, given how you use compareFn, you can insanely simplify it to:

```
function compareFn(a, b) {
  return a - b;
}
```

Reply:
Awesome thanks!